architecture/builds.adoc

```diff
@@ -0,0 +1,52 @@
+= Builds
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+A build is a process of transforming input parameters into resulting object.
+
+== BuildConfig
+`BuildConfig` object is the definition of the entire build process. It consists of
+following elements:
+* triggers - defines policies used for automatic invoking builds
+  * github webhook - github specific webhook, specifies which repository changes should
+    invoke new build, eg. each new commit triggers a new build. This trigger
+    is specific to github API.
+  * generic webhook - generic webhook is similar to github webhook in that it
+    invokes a new build whenever it gets a notification. The difference is its payload
+    which is slightly different than github's.
+  * image change - defines a trigger which will be invoked upon availability of
+    a new image in specified registry.
+* parameters
```

Comment:
indicate which of these are optional (source and output, i believe, since custombuilds need neither of those)